File: app/src/main/java/com/tevinjeffrey/rutgersct/adapters/holders/HeaderVH.java
package com.tevinjeffrey.rutgersct.adapters.holders;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;

public final class HeaderVH extends RecyclerView.ViewHolder {

    private final ViewGroup parent;

    public static HeaderVH newInstance(ViewGroup parent) {
        return new HeaderVH(parent);
    }

    private HeaderVH(ViewGroup itemView) {
        super(itemView);
        this.parent = itemView;
    }

    public void setHeaders(Iterable<View> headers) {
        for (View v : headers) {
            if (v.getParent() == null)
                parent.addView(v);
        }
    }
}

File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/utils/HeaderVH.kt
package com.tevinjeffrey.rutgersct.ui.utils

import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup

class HeaderVH private constructor(private val parent: ViewGroup) : RecyclerView.ViewHolder(parent) {

  fun setHeaders(headers: Iterable<View>) {
    headers
        .filter { it.parent == null }
        .forEach { parent.addView(it) }
  }

  companion object {

    fun newInstance(parent: ViewGroup): HeaderVH {
      return HeaderVH(parent)
    }
  }
}
